code/modules/mob/living/simple_animal/eldritch_demons.dm

```diff
@@ -61,11 +59,56 @@
 	maxHealth = 50
 	health = 50
 	sight = SEE_MOBS|SEE_OBJS|SEE_TURFS
-	spells_to_add = list(/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long,/obj/effect/proc_holder/spell/targeted/telepathy/eldritch)
+	spells_to_add = list(/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long,/obj/effect/proc_holder/spell/pointed/manse_link,/obj/effect/proc_holder/spell/targeted/telepathy/eldritch,/obj/effect/proc_holder/spell/pointed/trigger/blind/eldritch)
+
+	var/list/linked_mobs = list()
+	var/list/linked_actions = list()
+
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/Initialize()
+	. = ..()
+	link_mob(src)
 
 /mob/living/simple_animal/hostile/eldritch/raw_prophet/Login()
 	. = ..()
-	client?.view_size.setTo(11)
+	client?.view_size.setTo(10)
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/proc/link_mob(mob/living/mob_linked)
+	if(QDELETED(mob_linked) || mob_linked.stat == DEAD)
+		return FALSE
+	if(HAS_TRAIT(mob_linked, TRAIT_MINDSHIELD)) //mindshield implant, no dice
+		return FALSE
+	if(mob_linked.anti_magic_check(FALSE, FALSE, TRUE, 0))
+		return FALSE
+	if(mob_linked in linked_mobs)
+		return FALSE
+	linked_mobs.Add(mob_linked)
+	to_chat(mob_linked, "<span class='notice'>You feel something new enter your sphere of mind, you hear whispers of people far away, screeches of horror and a huming of welcome to [src]'s Mansus Link.</span>")
+	var/datum/action/innate/mansus_speech/action = new(src)
+	linked_actions.Add(action)
+	action.Grant(mob_linked)
+	RegisterSignal(mob_linked, list(COMSIG_MOB_DEATH, COMSIG_PARENT_QDELETING) , .proc/unlink_mob)
+	return TRUE
+
+/mob/living/simple_animal/hostile/eldritch/raw_prophet/proc/unlink_mob(mob/living/mob_linked)
+	var/link_id = linked_mobs.Find(mob_linked)
+	if(!(link_id))
+		return
+	UnregisterSignal(mob_linked, list(COMSIG_MOB_DEATH, COMSIG_PARENT_QDELETING))
+	var/datum/action/innate/mansus_speech/action = linked_actions[link_id]
+	action.Remove(mob_linked)
+	to_chat(mob_linked, "<span class='notice'>Your mind shatters as the [src]'s Mansus Link leaves your mind.</span>")
+	mob_linked.emote("Scream")
+	//micro stun
+	mob_linked.AdjustParalyzed(0.5 SECONDS)
+
+	linked_mobs[link_id] = null
+	linked_actions[link_id] = null
```

Comment:
This doesn't work like you expect it to.
`Find` returns you the numeric index of the list, which you can't use associatively.